plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/LocalAppEngineServerWizardFragment.java

```diff
@@ -24,46 +24,48 @@
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.jface.preference.PreferenceDialog;
+import org.eclipse.osgi.util.NLS;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.events.DisposeEvent;
 import org.eclipse.swt.events.DisposeListener;
 import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
 import org.eclipse.ui.dialogs.PreferencesUtil;
 import org.eclipse.wst.server.ui.wizard.IWizardHandle;
 import org.eclipse.wst.server.ui.wizard.WizardFragment;
 
-// TODO: Fix 'Edit' button
 public class LocalAppEngineServerWizardFragment extends WizardFragment {
-  private boolean cloudSdkConfigured;
+
+  private String cloudSdkPath = null;
   private boolean dialogFinished = false;
   private boolean enableFinishButton;
 
   public LocalAppEngineServerWizardFragment() {
-    cloudSdkConfigured = isCloudSdkConfigured();
+    cloudSdkPath = getCloudSdkLocation();
   }
 
   @VisibleForTesting
-  LocalAppEngineServerWizardFragment(boolean cloudSdkConfigured)
-  {
-    this.cloudSdkConfigured = cloudSdkConfigured;
+  LocalAppEngineServerWizardFragment(String cloudSdkPath) {
+    this.cloudSdkPath = cloudSdkPath;
   }
 
   @Override
   public boolean hasComposite() {
-    return !cloudSdkConfigured;
+    return true;
   }
 
   @Override
   public boolean isComplete() {
-    return cloudSdkConfigured || enableFinishButton;
+    return enableFinishButton || cloudSdkPath != null;
   }
 
   @Override
   public void enter() {
-    cloudSdkConfigured = isCloudSdkConfigured();
+    // TODO why do we check here AND in the constructor?
+    cloudSdkPath = getCloudSdkLocation();
```

Comment:
We called isCloudSdkConfigured (now getCloudSdkLocation) in the constructor, because we needed to determine if we had to create a composite or not.